src/docs/markdown/automatic-https.md

```diff
@@ -220,3 +222,84 @@ Before attempting any ACME transactions, Caddy will test the configured storage
 Caddy can obtain and manage wildcard certificates when it is configured to serve a site with a qualifying wildcard name. A site name qualifies for a wildcard if only its left-most domain label is a wildcard. For example, `*.example.com` qualifies, but these do not: `sub.*.example.com`, `foo*.example.com`, `*bar.example.com`, and `*.*.example.com`.
 
 To get a wildcard from Let's Encrypt, you simply need to enable the [DNS challenge](#dns-challenge) and use a wildcard domain in your config. We recommend using wildcards only when you have so many subdomains that you would encounter CA rate limits trying to obtain certificates for them all.
+
+
+
+## Examples
+
+Caddy implicitly uses the HTTPS port (default 443) for your [site addresses](/docs/conventions#network-addresses) that don't specify a port explicitly (which would disable automatic HTTPS). 
+
+Automatic HTTPS can be configured via the Caddyfile with the [`auto_https` global option](/docs/caddyfile/options), or via [per-server JSON configuration](/docs/json/apps/http/servers/automatic_https/). The Caddyfile option can be set to either of the following:
+- `disable_redirects` which disables the implicit HTTP->HTTPS redirect.
+- `off` which disables automatic HTTPS altogether, including the HTTP->HTTPS redirect and automatic enabling of TLS for sites that meet the requirements.
+
+| auto_https        | HTTP        | HTTPS       |
+|-------------------|-------------|-------------|
+| *default*         | redirects   | connects    |
+| disable_redirects | unreachable | connects    |
+| off               | connects    | unreachable |
```

Comment:
What conditions and configuration does this table assume? For example, here's a (contrived) Caddyfile where the table is not accurate:

```
{
    auto_https off
}

https://example.com

tls cert.pem key.pem
```

This will connect on port 443 but not 80. (Granted, this Caddyfile uses the `auto_https` option unnecessarily since providing my own cert and key disables it for that site anyway.)

Also a slight formatting suggestion, since `auto_https` refers to a config parameter:

```suggestion
| `auto_https`      | HTTP        | HTTPS       |
|-------------------|-------------|-------------|
| *default*         | redirects   | connects    |
| disable_redirects | unreachable | connects    |
| off               | connects    | unreachable |
```

Reply:
The `unreachable` and `connects` are defined beneath the table. They clarify that default without additional explicit configuration (such as a protocol scheme or port on a site address) `connects` is the implicit port/protocol used, and `unreachable` is just that, not implicitly reachable. `redirects` wasn't explained as I think that's been covered clearly that it is just redirecting HTTP connections to HTTPS to be handled.

> For example, here's a (contrived) Caddyfile where the table is not accurate

You didn't rely on the site address being implicitly configured for HTTP/HTTPS, so it's not valid. Granted I've only thought of it regarding ports.

There's an example for `disable_redirects` that uses the protocol schemes, but I personally had no idea how HTTPS is configured without the auto feature or what `auto_https` does "under the hood" equivalent to manual configuration. Traefik has a nice way with entrypoints and routers, the docs for Caddy probably touch on this somewhere, I just haven't gone through them in a linear fashion. 

Perhaps it's worth mentioning early on in the doc (via a link to where ever non-auto equivalents are explained) and maybe once more at the start of the examples section?

Does using an explicit port 443 also enable HTTPS(provided it matches the configured HTTPS port)? What does a user need to do to enable TLS / HTTPS on a non-standard port that's also not the standard HTTPS configured port(if changed from the default 443)?

> Also a slight formatting suggestion, since auto_https refers to a config parameter

The values in that column are also relevant to use inline code blocks then? (except for *default*, not sure if that's presented as a non-value, does it have any valid value like `on`?)